src/write.rs

```diff
@@ -466,7 +472,7 @@ impl<W: Write + io::Seek> ZipWriter<W> {
             options.permissions = Some(0o644);
         }
         *options.permissions.as_mut().unwrap() |= 0o100000;
-        self.start_entry(name, options)?;
+        self.start_entry(name, options, None)?;
```

Comment:
I think this change will switch the the writer too early resulting in compressed extra data. That's why I moved the switching out of `start_entry()` and did it explicitly.

Reply:
Great to know, thanks ^^ I'm expecting there to be a few conflicts with the other PRs that were hidden in the merge, and this will be good to keep in mind.

Btw if you'd rather keep #200 active, feel free to make this merge yourself. I just wasn't able to edit your PR

Reply:
Ok, I will keep #200 conflict-free and try to cover this case with a test. Enabling compression in `end_to_end.rs` should do it.

Reply:
Done, compression is enabled in the test by using `FileOptions::default()` for the file with extra data.